src/GraphQL.Tests/Execution/VariablesTests.cs

```diff
@@ -499,6 +552,34 @@ public void allows_non_nullable_inputs_to_be_set_to_a_value_directly()
 
             AssertQuerySuccess(query, expected);
         }
+
+        [Fact]
+        public void allows_custom_scalar_that_resolves_to_an_object()
+        {
+            var query = @"
+            query SetsCustomScalarInput($input: JsonScalarReturningObject) {
+              fieldWithCustomScalarInput(input: $input)
+            }
+            ";
+
+            var expected = @"
+            {
+              'fieldWithCustomScalarInput': ""bear-cat, dog, bird""
+            }
+            ";
+
+            var jsonString = new JObject
+            {
+                ["stringProperty"] = "bear",
+                ["arrayProperty"] = new JArray {"cat", "dog", "bird"}
+            }.ToString();
+
+            var inputs = new Inputs(new Dictionary<string, object>
+            {
+                ["input"] = jsonString
+            });
+            AssertQuerySuccess(query, expected, inputs);
```

Comment:
This is non-standard creation of `Inputs`.  All of the values should be converted from JSON.NET types to base types (string, int, etc.).  Have you ran this using the `.ToInputs` extension method on your input variables without your change?  Does that make it needed?

Reply:
@joemcbride I just did it that way to make the test easier to write. I've pushed a new commit that uses ToInputs.